applications/soupbintcp-gateway/src/main/java/com/paritytrading/nassau/soupbintcp/gateway/Events.java

```diff
@@ -76,15 +76,9 @@ public void receive() throws IOException {
 
                         toKeepAlive.add(session);
                     } else {
-                        Receiver receiver = (Receiver)key.attachment();
-
-                        try {
+                        try (Receiver receiver = (Receiver)key.attachment()) {
                             receiver.receive();
-                        } catch (IOException e1) {
-                            try {
-                                receiver.close();
-                            } catch (IOException e2) {
-                            }
+                        } catch (IOException e) {
```

Comment:
Is this doing the right thing? 🤔  `receiver.close()` should be invoked if an `IOException` occurs in the `receiver.receive()` invocation. To me, it looks like `receiver.close()` would get called unconditionally when exiting the `try` block.

Reply:
Right you are -- my mistake -- undoing that change.